example.py

```diff
@@ -106,13 +106,18 @@ def main(
 
 cheese =>""",
     ]
-    results = generator.generate(
-        prompts, max_gen_len=256, temperature=temperature, top_p=top_p
-    )
-
-    for result in results:
-        print(result)
-        print("\n==================================\n")
+    if max_batch_size==1:
```

Comment:
Hey! Good idea!
Could be a way of unifying the logic of both, thus cleaning the code?
I mean the following:
```python
for idx in range(0, len(prompts), max_batch_size):
    actual_prompts = prompts[idx:idx+max_batch_size]
    results = generator.generate(actual_prompts, max_gen_len=256, temperature=temperature, top_p=top_p)

    for result in results:
        print(result)
        print("\n==================================\n")
```

Reply:
this is great 🥇  thank you. I will add it